core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -396,6 +397,17 @@ protected PlanNodeStatsEstimate visitFunctionCall(FunctionCall node, Void contex
             return PlanNodeStatsEstimate.unknown();
         }
 
+        private Optional<Symbol> getSymbolReference(Expression expression)
+        {
+            if (expression instanceof SymbolReference) {
+                return Optional.of(Symbol.from(expression));
+            }
+            if (!(expression instanceof Cast) || !(((Cast) expression).getExpression() instanceof SymbolReference)) {
+                return Optional.empty();
+            }
+            return Optional.of(Symbol.from(((Cast) expression).getExpression()));
```

Comment:
unwrapping Cast here looks like an assumption that "stats representation" (eg ranges) before and after cast are comparable. I.e. that `io.trino.spi.statistics.StatsUtil#toStatsRepresentation` is the same (or almost the same) before and after the cast. 

This isn't true, however, is it?

Reply:
Yes, I've now added logic to do this unwrapping only if the cast is `isTypeOnlyCoercion`
PTAL and let me know if this is narrow enough to ensure correctness